core_lib/src/graphics/bitmap/bitmapimage.cpp

```diff
@@ -85,6 +85,21 @@ BitmapImage* BitmapImage::clone()
     return new BitmapImage(*this);
 }
 
+void BitmapImage::loadFile()
+{
+    mImage = std::make_shared<QImage>(fileName());
+    mBounds.setSize(mImage->size());
+    qDebug() << "Load file=" << fileName();
+}
+
+void BitmapImage::unloadFile()
+{
+    if (isModified() == false)
```

Comment:
Could we do setModified(false) when the image is saved and have a check in setModified that would reset mImage if the argument is false and the activeframepool does not contain the image? As it stands now, memory consumption is going to continue to increase for the lifetime of the program while making modifications, but if we did this, then all the old modified frames would removed from memory when saved.

Reply:
We do `setModified(false)` in `layerbitmap.cpp` line 78, in the `LayerBitmap::saveKeyFrameFile()` function. 
All modified frames will be written back to disk when auto-save is triggered. And we only do that with modified frames, so the auto-save will be a lot faster now.

Reply:
That's good then we don't have to change that. What I was mainly trying to say is that those frames that are written to disk should be unloaded if they are not in the active pool. BitmapImage::unloadFile is only called when it's removed from the activethreadpool, but it won't be removed from memory if it is in a modified state at that time. If after that, a save or an autosave writes those changes to disk and removes the modified state, the frame will remain loaded unnecessarily.

Reply:
right, any good suggestion for that?

Reply:
Well off the top of my head we could do something like this. Emit a modified signal when a keyframe is modified (from setModified/modification). Then:
```
void ActiveFramePool::unloadFrame(KeyFrame* key) {
    ...
    if(key->isModified()) {
        connect(key, &KeyFrame::modified, this, &ActiveFramePool::handler);
    }
    key->unloadFile();
}
```
The handler function would check if the new modified state is false and if the map does not contain the sender keyframe, and if these conditions are met it would call unloadFile again.